android/app/src/main/java/org/fossasia/openevent/viewmodels/LocationsFragmentViewModel.java

```diff
@@ -23,15 +27,42 @@ public LocationsFragmentViewModel() {
         realmRepo = RealmDataRepository.getDefaultInstance();
     }
 
-    public LiveData<List<Microlocation>> getLocations() {
+    public LiveData<List<Microlocation>> getLocations(String searchText) {
+        setSearchText(searchText);
+
+        if (filteredLocations == null) {
+            filteredLocations = new MutableLiveData<>();
+        }
+
         if (locations == null) {
             locations = new MutableLiveData<>();
             realmResults = realmRepo.getLocations();
             realmResults.addChangeListener((microlocations, orderedCollectionChangeSet) -> {
                 locations.setValue(microlocations);
+                getFilteredLocations();
             });
+        } else {
+            getFilteredLocations();
+        }
+        return filteredLocations;
+    }
+
+    private void getFilteredLocations() {
+        final String query = searchText.toLowerCase(Locale.getDefault());
+
+        List<Microlocation> filteredLocationsList = Observable.fromIterable(locations.getValue())
+            .filter(microlocation -> microlocation.getName()
+                .toLowerCase(Locale.getDefault())
+                .contains(query))
+            .toList().blockingGet();
```

Comment:
Great work. But don't do blocking get, you can subscribe and do the rest further